docs/core/howto/tap.rst

```diff
@@ -310,6 +310,107 @@ For a full list of cred plugins supported, see :api:`twisted.plugins <twisted.pl
 
 
 
+Deploy your application using python packages
+---------------------------------------------
+
+
+
+To deploy your application one possibility is to wrap it up in a python package.
+For this you need to write a special file ``setup.py``, which contains metadata
+of the package. You would have to extend the layout of your files like this:
+
+
+
+
+- **MyProject** - Top level directory
+
+
+- ``setup.py`` - Description file for the package
+
+
+  - **myproject** - Python package
+
+
+    - **__init__.py**
+
+
+  - **twisted**
+
+
+    - **plugins**
+
+      - ``myplugin.py`` - Plugin file
+
+
+
+
+
+.. code-block:: python
+
+
+    from distutils.core import setup
+
+    setup(
+        name='MyApplication',
+        version='0.1dev',
+        packages=['myproject', 'twisted.plugins'],
+        license='Creative Commons Attribution-Noncommercial-Share Alike license',
+        long_description="Insert your long description here",
+        install_requires=[
+            'twisted',
+            ],
+    )
```

Comment:
Perhaps it would be worth highlighting the part of this that's actually relevant to twistd plugins... The `twisted.plugins` entry in `packages`?

Also, does this really work?  For a long time I've had the impression that it does not.  Instead, obscure alternate options like `py_modules` are required.

Reply:
I am still waiting for someone with real distutils savvy to answer my question here: https://stackoverflow.com/questions/7275295/how-do-i-write-a-setup-py-for-a-twistd-twisted-plugin-that-works-with-setuptools

None of the answers seem really satisfying.

However, I just do the `packages` thing in the various packages I distribute nowadays - see for example https://github.com/glyph/txsni/blob/master/setup.py - and nobody seems to complain.  I don't know if this would cause issues if someone tried to build a Debian packages from these things or not, but increasingly it seems that building for distribution in the global package-manager-managed Linux site-packages is a Worst Practice.